atoma-state/src/state_manager.rs

```diff
@@ -1587,6 +1639,63 @@ impl AtomaState {
         Ok(maybe_stack)
     }
 
+    /// Locks (reserves) a specified number of compute units for a stack.
+    ///
+    /// This method atomically updates the `already_computed_units` field in the `stacks` table
+    /// by adding the specified number of compute units to the current value. This effectively
+    /// "locks" these compute units for use by preventing other processes from using the same units.
+    ///
+    /// # Arguments
+    ///
+    /// * `stack_small_id` - The unique small identifier of the stack to lock compute units for.
+    /// * `available_compute_units` - The number of compute units to lock/reserve.
+    ///
+    /// # Returns
+    ///
+    /// - `Result<()>`: A result indicating success (Ok(())) or failure (Err(AtomaStateManagerError)).
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The database query fails to execute.
+    /// - The specified stack is not found (`AtomaStateManagerError::StackNotFound`).
+    ///
+    /// # Example
+    ///
+    /// ```rust,ignore
+    /// use atoma_node::atoma_state::AtomaStateManager;
+    ///
+    /// async fn lock_units(state_manager: &AtomaStateManager) -> Result<(), AtomaStateManagerError> {
+    ///     let stack_small_id = 1;
+    ///     let units_to_lock = 100;
+    ///     
+    ///     state_manager.lock_compute_units_for_stack(stack_small_id, units_to_lock).await
+    /// }
+    /// ```
+    #[instrument(
+        level = "trace",    
+        skip_all,
+        fields(
+            %stack_small_id,
+            %available_compute_units
+        )
+    )]
+    pub async fn lock_compute_units_for_stack(
+        &self,
+        stack_small_id: i64,
+        available_compute_units: i64,
+    ) -> Result<()> {
+        let result = sqlx::query("UPDATE stacks SET already_computed_units = already_computed_units + $1 WHERE stack_small_id = $2")
```

Comment:
you are allowing to set `already_computed_units` above the `num_compute_units` of this stack

Reply:
Nice catch !

Reply:
push changes